docs/hooks.md

```diff
@@ -0,0 +1,108 @@
+# Hooks
+
+Cucumber provides a number of hooks which allow us to run blocks at various points in the Cucumber test cycle. You can define them anywhere in your support or step definition layers, using the methods `Before` and `After`. Hooks are global by default, meaning they run for every scenario in your
+features. If you want them to run for just certain scenarios, you need to tag
+those scenarios and then use a [tagged hook](hooks.md#tagged-hooks).
+
+## Scenario hooks
+
+`Before` hooks will be run before the first step of each scenario. They will run in the same order of which they are registered.
```

Comment:
Can you make a h3 section for Before and After hooks? can you also describe how Before hooks are similar to Background, and when to use each one? (Before hooks are for techynthings like starting browsers or servers, starting database transactions etc,